.github/workflows/ci.yml

```diff
@@ -0,0 +1,175 @@
+
+          # Configure test account
+          pgsql/bin/psql -U postgres -c "CREATE ROLE npgsql_tests SUPERUSER LOGIN PASSWORD 'npgsql_tests'"
+          pgsql/bin/psql -U postgres -c "CREATE DATABASE npgsql_tests OWNER npgsql_tests"
+        shell: bash
+
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1.0.2 # Issues with DOTNET_ROOT
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+
+      - name: Test (EFCore.PG.Tests)
+        run: dotnet test test/EFCore.PG.Tests --configuration Debug
+        shell: bash
+
+      - name: Test (EFCore.PG.FunctionalTests)
+        run: dotnet test test/EFCore.PG.FunctionalTests --configuration Debug
+        shell: bash
+
+      - name: Test (EFCore.PG.Plugins.FunctionalTests)
+        run: dotnet test test/EFCore.PG.Plugins.FunctionalTests --configuration Debug
+        shell: bash
+
+      - name: Pack NuGet packages (CI versions)
+        env:
+          GITHUB_SHA: ${{ github.sha }}
+        run: |
+          VERSION_SUFFIX="ci.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+          dotnet pack src/EFCore.PG          --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX
+          dotnet pack src/EFCore.PG          --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX --include-symbols
+          dotnet pack src/EFCore.PG.NodaTime --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX
+          dotnet pack src/EFCore.PG.NodaTime --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX --include-symbols
+          dotnet pack src/EFCore.PG.NTS      --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX
+          dotnet pack src/EFCore.PG.NTS      --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX --include-symbols
+        shell: bash
+
+      - name: Pack NuGet packages (Release versions)
+        run: |
+          dotnet pack src/EFCore.PG          --configuration Release --output nupkgs/release
+          dotnet pack src/EFCore.PG          --configuration Release --output nupkgs/release --include-symbols
+          dotnet pack src/EFCore.PG.NodaTime --configuration Release --output nupkgs/release
+          dotnet pack src/EFCore.PG.NodaTime --configuration Release --output nupkgs/release --include-symbols
+          dotnet pack src/EFCore.PG.NTS      --configuration Release --output nupkgs/release
+          dotnet pack src/EFCore.PG.NTS      --configuration Release --output nupkgs/release --include-symbols
+        shell: bash
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.os }}_dotnet${{ matrix.dotnet }}_pg${{ matrix.pg_major }}+sha.${{ github.sha }}
+          path: nupkgs
+
+  publish:
+    needs: build
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: true
+      matrix:
```

Comment:
Does this need to be a matrix at all? Why not make it a simple job? It also doesn't seem to need PG, and probably not dotnet either (whatever SDK is on the image should be fine for pushing).

I'd also make it run on Linux because why not :) Then you can remove `shell: bash` from everywhere :)

Reply:
> Does this need to be a matrix at all?

It's just so we could select the correct artifacts package. But if we refactor so packages are built in this step, then we can drop this matrix too.

> I'd also make it run on Linux because why not :)

Ugh, would like to too, but I ran into issues pushing to the GitHub Package Registry (GPR). You need `nuget.exe` to do it right now, and I couldn't get the mono version of `nuget` to work right (could be user error).